packages/react/src/analytics/integrations/shared/GoogleConsentMode/GoogleConsentMode.ts

```diff
@@ -0,0 +1,154 @@
+   */
+  private loadDefaults(initConsent: ConsentData | null) {
+    if (this.config) {
+      const initialValue: Record<string, string | number> = {};
+
+      if (this.config.wait_for_update) {
+        initialValue['wait_for_update'] = this.config.wait_for_update;
+      }
+
+      // Obtain default google consent registry
+      const consentRegistry = Object.keys(this.consentConfig).reduce(
+        (result, consentKey) => ({
+          ...result,
+          [consentKey]:
+            this.consentConfig[consentKey]?.default || GoogleConsentType.Denied,
+        }),
+        initialValue,
+      );
+
+      // Write default consent to data layer
+      this.write('consent', 'default', consentRegistry);
+
+      // write regions to data layer if they exists
+      this.config.regions?.forEach(region => {
+        this.write('consent', 'default', region);
+      });
+
+      this.updateConsent(initConsent);
+    }
+  }
+
+  /**
+   * Update consent.
+   *
+   * @param consentData - The consent data to be set.
+   */
+  updateConsent(consentData: ConsentData | null) {
+    if (this.config) {
+      // Dealing with null or undefined consent values
+      const consentList = consentData || {};
+
+      // Fill consent value into consent element, using analytics consent categories
+      const consentRegistry = Object.keys(this.consentConfig).reduce(
+        (result, consentKey) => {
+          let consentValue = GoogleConsentType.Denied;
+          const consent = this.consentConfig[consentKey];
+
+          if (consent) {
+            // has consent config key
+
+            if (consent.getConsentValue) {
+              // give priority to custom function
+              consentValue = consent.getConsentValue(consentList);
+            } else if (
+              consent?.categories !== undefined &&
+              consent.categories.every(consent => consentList[consent])
+            ) {
+              // The second option to assign value is by categories list
+              consentValue = GoogleConsentType.Granted;
+            }
+          }
+
+          return {
+            ...result,
+            [consentKey]: consentValue,
+          };
+        },
+        {},
+      );
+
+      // Write consent to data layer
+      this.write('consent', 'update', consentRegistry);
+    }
+  }
+
+  /**
+   * Handles consent by updating the data layer with consent information.
+   *
+   * @param consent - The consent command "consent".
+   * @param command - The command "default" or "update".
+   * @param consentParams - The consent arguments.
+   */
+  private write(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    consentCommand: 'consent',
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    command: 'default' | 'update',
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    consentParams:
+      | Record<string, Array<string> | string | number | undefined>
```

Comment:
Is this type correct? I think this is a Record<string, GoogleConsentType> type.

Reply:
We need to allow the extra possibilities like region or wait_for_update, which is an Array of strings and number.